.github/workflows/build_packages.yml

```diff
@@ -16,37 +30,51 @@ permissions:
   contents: read
 
 jobs:
-  # Note: metadata generation could happen in a separate trigger/schedule
-  # workflow. For cross platform builds, it's useful to just generate the
-  # metadata on Linux and pass that to later jobs using artifacts.
+  # Generate metadata on Linux and pass to later jobs.
   setup_metadata:
     if: ${{ github.repository_owner == 'nod-ai' || github.event_name != 'schedule' }}
     runs-on: ubuntu-24.04
     outputs:
-      shark_package_version: ${{ steps.version.outputs.shark_package_version }}
+      version_suffix: ${{ steps.version_rc.outputs.version_suffix }}
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - name: Setup Python
         uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
         with:
           python-version: 3.12
-          cache: "pip"
+
       - name: Install Python packages
+        run: pip install packaging
+
+      # Compute version suffix based on inputs (default to 'rc')
+      - name: Compute stable version suffix
+        if: ${{ inputs.build_type == 'stable' }}
         run: |
-          pip install packaging
-          pip freeze
-      - name: Generate release candidate versions
-        id: version_rc
+          version_suffix=""
+          echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
+      - name: Compute rc version suffix
+        if: ${{ inputs.build_type == 'rc' || inputs.build_type == '' }}
         run: |
           version_suffix="$(printf 'rc%(%Y%m%d)T')"
           echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
-          sharktank_package_version=$(python3 build_tools/python_deploy/compute_local_version.py --version-suffix=${version_suffix} sharktank)
-          shortfin_package_version=$(python3 build_tools/python_deploy/compute_local_version.py --version-suffix=${version_suffix} shortfin)
-          sharkai_package_version=$(python3 build_tools/python_deploy/compute_common_version.py -rc --version-suffix=${version_suffix} --write-json)
-      - name: Upload version_local.json
+      - name: Compute dev version suffix
+        if: ${{ inputs.build_type == 'dev' }}
+        run: |
+          version_suffix=".dev0+${{ github.sha }}"
+          echo "version_suffix=${version_suffix}" >> $GITHUB_ENV
+
+      - name: Generate release candidate versions
+        id: version_rc
+        run: |
+          echo "version_suffix=${version_suffix}" >> $GITHUB_OUTPUT
+          python3 build_tools/python_deploy/compute_local_version.py --version-suffix=${version_suffix} sharktank
+          python3 build_tools/python_deploy/compute_local_version.py --version-suffix=${version_suffix} shortfin
+          python3 build_tools/python_deploy/compute_common_version.py -rc --version-suffix=${version_suffix} --write-json
```

Comment:
I think this works (as the version suffix is empty for stable releases and overwrites for dev releases) but we should refactor in a follow up (happy to take this) as it rather confusing that `-rc` is passed for all builds. I needed to look twice here to understand why it is probably fine.

Reply:
Oh, I missed the `-rc` in the middle there. Yeah, it still works but we should refactor :P

Reply:
This is addressed with #674.